tests/test_temporal.py

```diff
@@ -2356,3 +2377,754 @@ async def test_beta_graph_parallel_execution_in_workflow(client: Client):
+        provider_factory_called = True
+        raise AssertionError('provider_factory should not be called when model_id is None')
+
+    temporal_agent = TemporalAgent(
+        agent,
+        name='resolve_reinfer_default',
+        provider_factory=mock_provider_factory,
+    )
+
+    temporal_model = temporal_agent._temporal_model  # pyright: ignore[reportPrivateUsage]
+
+    infer_model_called = False
+
+    def mock_infer_model(
+        model_name: str, provider_factory: Callable[[str], Provider[Any]]
+    ) -> TestModel:  # pragma: no cover
+        nonlocal infer_model_called
+        infer_model_called = True
+        raise AssertionError('infer_model should not be called when model_id is None')
+
+    monkeypatch.setattr('pydantic_ai.durable_exec.temporal._model.models.infer_model', mock_infer_model)
+
+    # model_id is None - should return the wrapped model
+    params = _RequestParams(
+        messages=[],
+        model_settings=None,
+        model_request_parameters=ModelRequestParameters(),
+        serialized_run_context=None,
+        model_id=None,
+    )
+
+    result = temporal_model._resolve_model(params, {'key': 'value'})  # pyright: ignore[reportPrivateUsage]
+    # When model_id is None, returns the wrapped model directly
+    assert result is test_model
+    # Verify neither was called
+    assert not provider_factory_called
+    assert not infer_model_called
+
+
+def test_temporal_model_request_without_run_context(monkeypatch: MonkeyPatch):
+    """Test request() when run_context is None (covers branch 161->165)."""
+    test_model = TestModel()
+    agent = Agent[None, str](test_model, name='request_no_context')
+    temporal_agent = TemporalAgent[None, str](agent, name='request_no_context')
+
+    temporal_model = temporal_agent._temporal_model  # pyright: ignore[reportPrivateUsage]
+
+    # Track what gets passed to execute_activity
+    activity_calls: list[tuple[Any, ...]] = []
+
+    async def mock_execute_activity(activity: Any, args: list[Any], **kwargs: Any) -> ModelResponse:
+        activity_calls.append((activity, args, kwargs))
+        return ModelResponse(parts=[TextPart(content='test response')])
+
+    # Mock workflow.in_workflow() to return True, but don't set CURRENT_RUN_CONTEXT
+    monkeypatch.setattr('pydantic_ai.durable_exec.temporal._model.workflow.in_workflow', lambda: True)
+    monkeypatch.setattr('pydantic_ai.durable_exec.temporal._model.workflow.execute_activity', mock_execute_activity)
+
+    asyncio.get_event_loop().run_until_complete(
+        temporal_model.request(
+            messages=[],
+            model_settings=None,
+            model_request_parameters=ModelRequestParameters(),
+        )
+    )
+
+    # Verify execute_activity was called
+    assert len(activity_calls) == 1
+    _, args, _ = activity_calls[0]
+    params = args[0]
+    deps = args[1]
+
+    assert params.serialized_run_context is None
+    # deps should also be None
+    assert deps is None
```

Comment:
I didn't review the remaining tests, I think we should be able to remove the vast majority of these and only have tests with code as a the user would write it, that verify it works as expected